runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -47,13 +47,22 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
             if !polling_state.azure_async_operation_header_link.nil?
               update_state_from_azure_async_operation_header(polling_state.get_request(headers: request.headers, base_uri: request.base_uri, user_agent_extended: user_agent_extended), polling_state)
             elsif !polling_state.location_header_link.nil?
-              update_state_from_location_header(polling_state.get_request(headers: request.headers, base_uri: request.base_uri, user_agent_extended: user_agent_extended), polling_state, custom_deserialization_block)
+              update_state_from_location_header(polling_state.get_request(headers: request.headers, base_uri: request.base_uri, user_agent_extended: user_agent_extended), polling_state, custom_deserialization_block, final_state_via)
```

Comment:
do we need to be careful with order of parameters due to the deserialization block parameter? should that be the last one? not sure if there are restrictions for blocks - found something here, not sure if it helps https://stackoverflow.com/questions/38614973/ruby-method-with-optional-options-and-block-parameter

Reply:
Agreed that if it is a block parameter it should be last. But in this case, though it is called custom deserialization block, it is not the traditional block parameter of ruby that is indicated with '&' in method signature. So, adding the optional parameter at the end should be fine (Also it works with the existing code) :-)

Reply:
sounds good, thx!